apiserver/pkg/common/common.go

```diff
@@ -91,3 +92,40 @@ func GetAPIServer(ctx context.Context, dynamicClient dynamic.Interface, external
 	}
 	return api, nil
 }
+
+func GetObjStatus(obj client.Object) string {
+	if obj == nil {
+		return ""
+	}
+	if obj.GetDeletionTimestamp() != nil {
+		return "Deleting"
+	}
+
+	var (
+		condition v1alpha1.Condition
+	)
+	switch obj.GetObjectKind().GroupVersionKind().Kind {
+	case "Datasource":
+		v := obj.(*v1alpha1.Datasource)
+		condition = v.Status.GetCondition(v1alpha1.TypeReady)
+	case "Embedder":
+		v := obj.(*v1alpha1.Embedder)
+		condition = v.Status.GetCondition(v1alpha1.TypeReady)
+	case "KnowledgeBase":
+		v := obj.(*v1alpha1.KnowledgeBase)
+		condition = v.Status.GetCondition(v1alpha1.TypeReady)
+	case "LLM":
+		v := obj.(*v1alpha1.LLM)
+		condition = v.Status.GetCondition(v1alpha1.TypeReady)
+	case "Model":
+		v := obj.(*v1alpha1.Model)
+		condition = v.Status.GetCondition(v1alpha1.TypeReady)
+	case "Worker":
+		v := obj.(*v1alpha1.Worker)
+		condition = v.Status.GetCondition(v1alpha1.TypeReady)
```

Comment:
@0xff-dev  For worker,we use `reason` as `status` to provide fine-grained  status